text/entities/0264-resource-and-entities.md

```diff
@@ -0,0 +1,284 @@
+
+Going forward, we have set of options:
+
+- Duplicate attributes in `Entity` section of Resource.
+- Reference attributes of Resource in entity.
+- Only identify Entity id and keep attribute<->entity association out of band.
+- Extend Attribute on Resource so that we can track the entity type per Key-Value (across any attribute in OTLP).
+
+The third option prevents generic code from interacting with Resource and Entity without understanding the model of each.  The first keeps all usage of entity simple at the expense of duplicating information and the middle is awkward to interact with from an OTLP usage perspective. The fourth is violates our stability policy for OTLP.
+
+### How to deal with Resource/Entities whose lifecycle does not match the SDK?
+
+This proposal motivates a Resource Coordinator in the SDK whose job could include managing changes in entity lifetimes, but does not account for how these changes would be broadcast across TracerProvider, LogProvider, MeterProvider, etc.  That would be addressed in a follow on OTEP.
+
+### How to deal with Prometheus Compatibility for non-SDK telemetry?
+
+Today, Prometheus compatibility relies on two key attributes in Resource: `service.name` and `service.instance.id`. These are not guaranteed to exist outside of OpenTelemetry SDK generation. While this question is not fully answered, we believe outlining identity in all resources within OpenTelemetry allows us to define a solution in the future while preserving compatibility with what works today.
+
+A quick proposal of what this might look like:
+
+- `target_info` metric generation is updated to exclude any keys which are contained in `descriptive_attributes_keys` of an entity.
+- For each entity which has non-empty descriptive_attributes_keys, generate an info metric: `<entity_type>_entity_info` (naming TBD), which has all identifying and descriptive keys. This should play nicely with the planned improvements to [info-typed metrics](https://github.com/prometheus/proposals/blob/main/proposals/2024-04-10-native-support-for-info-metrics-metadata.md#goals).
+
+### Should entities have a domain?
+
+Is it worth having a `domain` in addition to type for entity?  We could force each entity to exist in one domain and leverage domain generically in resource management.  Entity Detectors would be responsible for an entire domain, selecting only ONE to apply a resource. Domains could be layered, e.g. a Cloud-specific domain may layer on top of a Kubernetes domain, where "GKE cluster entity" identifies *which* kubernetes cluster a kuberntes infra entity is part of.  This layer would be done naively, via automatic join of participating entities or explicit relationships derived from GKE specific hooks.
```

Comment:
I think `Type` values should be namespaced like attribute values. For example Kubernetes Pods will have `Type` value of "k8s.pod". "k8s" is the namespace here, just like it is for relevant Kubernetes attributes. I think namespacing is sufficient for name conflict avoidance, we don't need an additional, separate `domain` field.

Reply:
Ack and I agree with you.

>E.g. could you rely on {foo}.{other} and {foo}.{other2} meaning that other + other2 are related in some way?

We don't currently formally define this meaning in any way, but that vague meaning is there. We don't just randomly put things in namespaces, we intuitively group them by _some_ characteristic. `k8s.pod.*` attributes `k8s.node.*` attributes don't just use `k8s.` prefix by chance, it is because pod and node are both Kubernetes concepts.

I don't know if we want to have a stricter definition of what it means for the type to be in the same namespace, but I think you have a good point: we should consider it.